Ejercicio EcommerceSite

ijchavez opened the pull request:
Agrego los primeros test con el PageObject realizado:
Copia de lo que te mande por mail:

ya tenia hechas muchas cosas de las pages que te fui preguntando en la semana y le agregue la clase que hicimos hoy MyPersonalInformationPage.java y pense todo el set de clases de la manera que te lo adjunto.

Yo registerToSite lo tengo en una clase de utilities y si bien el test anda siento que la ida y venida de las clases, test y pages a esa clase de utilities no me parece muy prolijo que digamos.

Tengo más que nada esa duda de diseño porque siento que es desordenado. Recièn en la clase comentaste que registerToSite podria ir en BaseTest pero me gustaria, si podes, que mires lo que te mando a ver si esta bien pensado o como podria pulir el ejercicio.

ijchavez committed:
Ejercicio EcommerceSite

Agrego los primeros test con el PageObject realizado:
Copia de lo que te mande por mail:

ya tenia hechas muchas cosas de las pages que te fui preguntando en la semana y le agregue la clase que hicimos hoy MyPersonalInformationPage.java y pense todo el set de clases de la manera que te lo adjunto.

Yo registerToSite lo tengo en una clase de utilities y si bien el test anda siento que la ida y venida de las clases, test y pages a esa clase de utilities no me parece muy prolijo que digamos.

Tengo más que nada esa duda de diseño porque siento que es desordenado. Recièn en la clase comentaste que registerToSite podria ir en BaseTest pero me gustaria, si podes, que mires lo que te mando a ver si esta bien pensado o como podria pulir el ejercicio.

seleniumcurso commented on ecommerceSite/Utilities.java:
```diff
@@ -0,0 +1,135 @@
+
+    }
+    //esto me parecio ponerlo aca y no en la clase de constantes ya que es una utilidad que se puede usar en otras paginas
+    public String getFakerName(){
```
podria ir en una clase DataFactory o FakeDataGenerator

ijchavez replied:
Listo Emiliano, ya la agregué pero si subo los cambios los hago ni bien tenga el test de la compra del vestido. Muchas gracias por tu tiempo

ijchavez replied:
Hecho y subido

seleniumcurso commented:
Gerardo, veo un excelente trabajo! solo un comentario para lo de los Fakers. El resto, lo veo sin errores.

ijchavez committed:
Agregado de Test BuyADress

Agregue el test de la compra del vestido me falta el paso 5 nada mas.
Mi duda es que al clickear sobre un vestido a comprar a veces me salta el pop up y a veces va directamente al producto. Yo lo tengo hecho el test para que vaya al producto entonces si sale por el camino de ir al pop up (como si clickeara en quickview) el test falla.
Esa es mi unica duda.

ijchavez commented:
> Gerardo, veo un excelente trabajo! solo un comentario para lo de los Fakers. El resto, lo veo sin errores.

Gracias Emiliano. Subi el test de la compra del vestido pero lo hice de manera propia sin ver tanto el ejercicio que hiciste. El unico problema que tengo es que al tocar en un vestido yo cree el test para que clickee en el producto y toque add to cart. 
El problema se da es que a veces se va por un boton que se llama Quick View y abre un pop up rompiendo el test. Hay manera de salvar esto o hacer que si va por un lado lo trate de una manera y si se va por otro lo trate de otra?

ijchavez committed:
test buyADress Completo

Agregue el test para comprar un vestido, asserts y demas.
Me queda pendiente la duda sobre que al hacer click sobre el vestido si me clickea en el quick view me abre un pop up en vez de una nueva pagina donde me muestra el vestido. Si eso no pasa el test navega haciendo la compra y llegando hasta la orden.

seleniumcurso commented:
> > Gerardo, veo un excelente trabajo! solo un comentario para lo de los Fakers. El resto, lo veo sin errores.
> 
> Gracias Emiliano. Subi el test de la compra del vestido pero lo hice de manera propia sin ver tanto el ejercicio que hiciste. El unico problema que tengo es que al tocar en un vestido yo cree el test para que clickee en el producto y toque add to cart.
> El problema se da es que a veces se va por un boton que se llama Quick View y abre un pop up rompiendo el test. Hay manera de salvar esto o hacer que si va por un lado lo trate de una manera y si se va por otro lo trate de otra?

Hola! seria bueno verlo en una llamada para que sea mas facil de visualizar y ver q se puede hacer

ijchavez commented:
> > > Gerardo, veo un excelente trabajo! solo un comentario para lo de los Fakers. El resto, lo veo sin errores.
> > 
> > 
> > Gracias Emiliano. Subi el test de la compra del vestido pero lo hice de manera propia sin ver tanto el ejercicio que hiciste. El unico problema que tengo es que al tocar en un vestido yo cree el test para que clickee en el producto y toque add to cart.
> > El problema se da es que a veces se va por un boton que se llama Quick View y abre un pop up rompiendo el test. Hay manera de salvar esto o hacer que si va por un lado lo trate de una manera y si se va por otro lo trate de otra?
> 
> Hola! seria bueno verlo en una llamada para que sea mas facil de visualizar y ver q se puede hacer

Emiliano, cuando creas conveniente lo podemos ver. Intente jugar on el isDisplayed() buscando la X de cierre clickear en quick view y hacer un if donde si se hacia ese display = true lo cerrara, pero no me funciono. 
Gracias!!

ijchavez committed:
Cambios finales a BuyADress

Le habia agregado el tema que explicaste hoy de JavascriptExecutor que explicaste porque lo busque en internet a ver como podía replicar el caso
Más que nada porque en las charlas por zoom que tuvimos luego de hacer el try/catch me di cuenta que cuando entraba el test por donde yo queria la pagina bajaba levemente. 
Por lo que cree sin saber con la información "window.scrollBy(0,200)" porque queria que fuera hasta la mitad de la pantalla.
Quedó de la siguiente manera en la clase DressPage.java:
    public DressesPage(WebDriver remoteDriver){
        driver = remoteDriver;
        //En cada constructor debo ponerlo, mas si hago extends de clases y lo van a tomar varios
        PageFactory.initElements(driver, this);
        JavascriptExecutor js = (JavascriptExecutor) driver;
        js.executeScript("window.scrollBy(0,200)");
    }

Pero no sé si lo puse donde debería ir.

ijchavez committed:
Clases del curso al dia 15/5

ijchavez committed:
Delete Intermedio directory
